### DISCUSSION TITLE
Projections working in terms of memory

### DISCUSSION THREAD

**Discussion by Shreya5PGit** [ClickHouse/ClickHouse#64627] [Q&A]
Hi i have created a projection based on my table. But as the documentation suggests projections in itself do consume space and IO how and where can i see the space consumed by my projections. I tried running a few queries on my systems table but did not get the projections space utilization separately.
Or can someone explain the memory and infra level changes that occur on the database or table on its creation on top of our table while creating a projection.

**Comment by den-crane** [comment]
system.projection_parts
system.projection_parts_columns

```sql

SELECT
    database,
    table,
    name,
    formatReadableSize(sum(data_compressed_bytes) AS size) AS compressed,
    formatReadableSize(sum(data_uncompressed_bytes) AS usize) AS uncompressed,
    round(usize / size, 2) AS compr_rate,
    sum(rows) AS rows,
    count() AS part_count
FROM system.projection_parts
WHERE (table = '....') AND active
GROUP BY
    database,
    table,
    name
ORDER BY size DESC;


SELECT
    database,
    table,
    column,
    formatReadableSize(sum(column_data_compressed_bytes) AS size) AS compressed,
    formatReadableSize(sum(column_data_uncompressed_bytes) AS usize) AS uncompressed,
    round(usize / size, 2) AS compr_rate
FROM system.projection_parts_columns
WHERE (active = 1) AND (table LIKE '....')
GROUP BY
    database,
    table,
    column
ORDER BY size DESC;

**Reply by Shreya5PGit** [reply]
Thanks ! But can you give a brief about its memory working and distribution based on the primary table? I want to understand exactly how do projections work in terms of memory and I/o on top of our original table

**Reply by Shreya5PGit** [reply]
Like my projection metrics are 
 --- Table ---                                                     -compres-  -uncompres- 
5PAppLogging1001_bkp  logging_projection  2.33 GiB       3.80 GiB

and table metrics
5PAppLogging1001_bkp                                   2.32 GiB      3.76 GiB

So does my projection entirely consume this much space or is it somewhat within the existing table memory.

**Reply by den-crane** [reply]
Depends on the projection, but it literally stores the result of the projection query. So if you order by projection and use `select *` projection can use even more space that the source data, plus it can mulitply i/o several times.

**Reply by Shreya5PGit** [reply]
Oh got it. Thank you Denny!
